### ISSUE TITLE
the find_jar_files is making a mess and the while id considering it all as a single line

### ISSUE THREAD

**Issue by louigi600** [rubo77/log4j_checker_beta#19]
I worked around 2 problems:
1) some of my servers have no internet connection so I need to be able to work on a cached vulnerable.hashes
2)the one line thing I worked around with an array
there are the workarounds:

```
dir_temp_hashes=$(mktemp -d --suffix _log4jscan)
file_temp_hashes="$dir_temp_hashes/vulnerable.hashes"
if [ -f vulnerable.hashes -a -s vulnerable.hashes ]
then
  ok_hashes=1
  cp vulnerable.hashes $dir_temp_hashes
  information "using cached vulnerable.hashes "
else
  information "no vulnerable.hashes cached ... downloading"
ok_hashes=
if [[ -n $SHA256_HASHES_URL && $(command -v wget) ]]; then
  wget  --max-redirect=0 --tries=2 -O "$file_temp_hashes.in" -- "$SHA256_HASHES_URL"
elif [[ -n $SHA256_HASHES_URL && $(command -v curl) ]]; then
  curl --globoff -f "$SHA256_HASHES_URL" -o "$file_temp_hashes.in"
fi
if [[ $? = 0 && -s "$file_temp_hashes.in" ]]; then
  cat "$file_temp_hashes.in" | cut -d" " -f1 | sort | uniq  > "$file_temp_hashes"
  ok_hashes=1
  information "Downloaded vulnerable hashes from $SHA256_HASHES_URL"
  cp $file_temp_hashes .
fi
fi
```

and the other thing about the one line thing from find_jar_files

  ```
JarArray=($(find_jar_files))
  #  while read -r jar_file; do
    for ((i=0;i<${#JarArray[*]};i++))
    ....
    ....
    ....
      done
#  done <<<$(find_jar_files)
```
  
Sorry I've no time to make a patch file or to clone edit and suggest a pull

**Comment by rubo77** [comment]
maybe use quotes:

    JarArray="$(find_jar_files)"

**Comment by rubo77** [comment]
Wat are you trying to achieve?

the script doesn't need internet connection. 

if you want to change the script so it works locally, just change the lines, where it downloads the latest hash file and use a local file instead

**Comment by rubo77** [comment]
Ah I get it, you are reporting a bug here :)

Thanks for the pr #20

**Comment by rubo77** [comment]
fixed by #20
